gravityforms/gravityformsakismet
# Refactor and update Akismet addon

## Description
This PR updates and refactors the Akismet addon in preparation for its release. Specifically it:

- Refactors several methods for more simplicity
- Breaks out the fields filter and plugin settings into their own classes
- Adds our unit testing framework and includes several tests
- Adds the menu icon
- Adds our namespace

This would resolve https://github.com/gravityforms/backlog/issues/1359. 

https://github.com/gravityforms/backlog/issues/1392 will be merged into this branch unless this is approved and merged into master first.

## Testing Instructions
- This is mostly a refactor so there isn't anything new to test, but we should make sure everythings still works
- Install and activate Akismet and this version of the Akismet Addon
- Go to Forms->Settings and toggle Akismet on
- Use a form with all fields or create a form with at least a name, email, url, and paragraph field
- Go to the form's settings and go to Akismet
- Make sure that default merge tags are populated
- You won't be able to add new merge tags for author, url, or email, but that will be fixed in Jeremy's branch when these are replaced with field maps
- Save the settings
- Preview and submit the form and make sure it goes through
- Preview again and submit the form with "viagra-test-123" in the name field. Make sure it's marked as spam

## Checklist
- [x] I've tested my code locally to ensure it functions as intended.
- [ ]  I've added a changelog entry (if necessary).
- [x]  My code passes all Unit Tests.
- [ ]  My code passes all Acceptance Tests.
- [x]  This PR has a related issue.
- [x]  This PR has a related test.
